extensions/binding/org.eclipse.smarthome.binding.lirc/src/main/java/org/eclipse/smarthome/binding/lirc/handler/LIRCRemoteHandler.java

```diff
@@ -0,0 +1,178 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.lirc.handler;
+
+import static org.eclipse.smarthome.binding.lirc.LIRCBindingConstants.CHANNEL_BUTTON;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.binding.lirc.internal.LIRCMessageListener;
+import org.eclipse.smarthome.binding.lirc.internal.config.LIRCRemoteConfiguration;
+import org.eclipse.smarthome.binding.lirc.internal.messages.LIRCButtonEvent;
+import org.eclipse.smarthome.binding.lirc.internal.messages.LIRCResponse;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LIRCRemoteHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Andrew Nagle - Initial contribution
+ */
+public class LIRCRemoteHandler extends BaseThingHandler implements LIRCMessageListener {
+
+    private Logger logger = LoggerFactory.getLogger(LIRCRemoteHandler.class);
+
+    private static final Pattern UNKNOWN_REMOTE_PATTERN = Pattern.compile("^unknown remote: \"(.+)\"$");
+
+    private LIRCBridgeHandler bridgeHandler;
+
+    private LIRCRemoteConfiguration config;
+
+    private String remoteName = null;
+
+    public LIRCRemoteHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        if (remoteName == null) {
+            logger.error("Remote name is not set in {}", getThing().getUID());
+            return;
+        }
+
+        if (channelUID.getId().equals(CHANNEL_BUTTON)) {
+            // TODO: handle command
+
+            if (command instanceof RefreshType) {
+                // not supported
+            } else if (command instanceof StringType) {
+                bridgeHandler.transmit(remoteName, command.toString());
+            }
+
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing thing {}", getThing().getUID());
+
+        config = getConfigAs(LIRCRemoteConfiguration.class);
+        remoteName = config.getRemote();
+
+        if (remoteName == null) {
+            logger.error("Remote name is not set in {}", getThing().getUID());
+            return;
```

Comment:
you MUST guarantee that the Thing's status is set sooner or later. If simply return here, this will not happen and it's going to stay in `INITIALIZING` forever. 

Therefore, e.g. 

```java
updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION ERROR, "Remote name is not set")
```

would be a sensible choice, like you did in `initializeBridge()`